src/hd.rs

```diff
@@ -314,49 +257,39 @@ impl HDWallet {
 
         (start_index..=end_index)
             .map(|index| {
-                let key = self.derive(index);
-                let address = self.to_address(&key);
-                let origin = KeyOrigin::Derived(key.parent_fingerprint, index);
-                (address, rescan_since, origin.to_label())
+                let label = format!("{}/{}", self.descriptor.checksum(), index);
```

Comment:
The FIXME is line 348 is because the label is getting formatted here without the `LABEL_PREFIX`. This should be changed to construct a `KeyOrigin` and use `to_label()` instead of manually crafting the string.